### ISSUE TITLE
Router hardware requirements

### ISSUE THREAD

**Issue by s0rl0v** [wiktorbgu/amneziawg-mikrotik#1]
Hi @wiktorbgu 

Thank you for this amazing project!

Could you also provide router hardware requirements? Will it work on hAP ac2 or it needs more beefy hardware?

WBR,
Sergey

**Comment by wiktorbgu** [comment]
Hello @s0rl0v!
Any mikrotik router that supports containers and has enough resources to run them, which can be verified experimentally, will do.  
On your hAP ac2, you can install containers on an external USB memory.

**Comment by rozhkovets** [comment]
Thanks for the project
Everything worked fine on the HAP AC 3, but slowly. The maximum I got was 25-30 Mbps. Apparently, software limitations have more of an effect than hardware limitations. In the logs [!] Missing WireGuard (Amnezia VPN) kernel module. Falling back to slow userspace implementation.

Спасибо за проект
На хап ац 3 все отлично заработало, но медленно. Максимум что я получил 25-30 Мегабит. Видимо больше влияюет програмные, а не аппаратные ограничения. В логах [!] Missing WireGuard (Amnezia VPN) kernel module. Falling back to slow userspace implementation.

**Comment by wiktorbgu** [comment]
No, this limitation depends more on the VPN than on the Mikrotik.
You probably connected to Warp, I also got this limitation. 
Just try another VPN, for example, Proton, and you will see the speed is several times higher.

Нет, это ограничение зависит скорее от vpn, нежели от самого микротика.
Вы наверное подключились к warp, я тоже получал такое ограничение. 
Просто попробуйте другой впн, например, proton и увидите скорость в разы больше.

**Comment by s0rl0v** [comment]
@rozhkovets Hi!
Could you share your RAM usage without amnezia-wg container and after?

**Comment by rozhkovets** [comment]
> @rozhkovets Hi! Could you share your RAM usage without amnezia-wg container and after?

without 87, after 66 MB

**Comment by avokain** [comment]
> No, this limitation depends more on the VPN than on the Mikrotik. You probably connected to Warp, I also got this limitation. Just try another VPN, for example, Proton, and you will see the speed is several times higher.
> 
> Нет, это ограничение зависит скорее от vpn, нежели от самого микротика. Вы наверное подключились к warp, я тоже получал такое ограничение. Просто попробуйте другой впн, например, proton и увидите скорость в разы больше.
> 
> Вы конечно можете увидеть скорости намного выше на том же самом vpn при подключении с компьютера или из самого микротика, но почему-то возникает такое явление и замечал я это только с warp'ом.

@wiktorbgu 

I also tested it on HAP AC3 and it's also 30 Mbit and the CPU is about 80-100%. As a server, classic WG, not WARP

Тоже потестировал на HAP AC3 и тоже 30 мбит и CPU около 80-100%. В качестве сервера, классический WG, не WARP

**Comment by wiktorbgu** [comment]
Use this to fix slow download speed

Используйте это для исправления низкой скорости загрузки

`PreUp=iptables -A FORWARD -p tcp --tcp-flags SYN,RST SYN -j TCPMSS --clamp-mss-to-pmtu`

**Comment by avokain** [comment]
Спасибо. Так намного лучше открываются сайты, без задержек. Но процессор все равно упирается в 100% на 30 мбит. Думаю это изза отсутствия аппаратного ускорения WG при работе через контейнер
